dusty/systems/rsync/__init__.py

```diff
@@ -54,11 +54,19 @@ def sync_repos(repos):
         log_to_client('Syncing {} repo {} to remote at {}'.format(repo_type, repo.remote_path, repo.vm_path))
         sync_local_path_to_vm(repo.local_path, repo.vm_path)
 
-def sync_repos_by_app_name(app_names):
+# For the bottom two functions to work, they really need to take already expanded specs
+def sync_repos_by_app_name(expanded_specs, app_names):
     repos = set()
-    assembled_specs = get_assembled_specs()
     for app_name in app_names:
-        for lib_name in assembled_specs['apps'][app_name].get('depends', {}).get('libs', []):
+        for lib_name in expanded_specs['apps'][app_name].get('depends', {}).get('libs', []):
             repos.add(get_repo_of_app_or_library(lib_name))
         repos.add(get_repo_of_app_or_library(app_name))
     sync_repos(repos)
+
+def sync_repos_by_lib_name(expanded_specs, lib_names):
+    repos = set()
+    for lib_name in lib_names:
+        for depends_lib_name in expanded_specs['libs'][lib_name].get('depends', {}).get('libs', []):
+            repos.add(get_repo_of_app_or_library(depends_lib_name))
+        repos.add(get_repo_of_app_or_library(lib_name))
+    sync_repos(repos)
```

Comment:
While you're here might as well rewrite these to remove all the duplication :shrug: